sklearn/feature_extraction/hashing.py

```diff
@@ -113,11 +119,59 @@ def _validate_params(n_features, input_type):
             raise ValueError("input_type must be 'dict', 'pair' or 'string',"
                              " got %r." % input_type)
 
-    def fit(self, X=None, y=None):
-        """No-op.
+    def _transform(self, raw_X, save_mappings):
+        raw_X = iter(raw_X)
+        if self.input_type == "dict":
+            raw_X = (_iteritems(d) for d in raw_X)
+        elif self.input_type == "string":
+            raw_X = (((f, 1) for f in x) for x in raw_X)
+
+        indices, indptr, values, feature_to_index_map_ = \
+            _hashing.transform(raw_X, self.n_features, self.dtype,
+                               self.alternate_sign, self.save_mappings)
```

Comment:
You're passing the wrong variable here. Which means you don't have a test to check the mappings are unchanged when transform is called